cibyl/orchestrator.py

```diff
@@ -208,68 +207,63 @@ def run_query(self, start_level=1):
                     # associated, we should not consider it here, e.g.
                     # --sources
                     continue
-                # the validation process provides a flat list of systems
-                # because the environment information is not used from this
-                # point forward
-                for system in valid_systems:
-                    if not system.is_enabled():
-                        continue
-                    system_name = system.name.value
-                    source_methods_store = system_methods_stores[system_name]
-                    # collect system-level arguments that can affect the
-                    # result of the source method call
-                    system_args = system.export_attributes_to_source()
+                if not system.is_enabled():
+                    return
+                source_methods_store = system_methods_stores[system.name.value]
+                # collect system-level arguments that can affect the
+                # result of the source method call
+                system_args = system.export_attributes_to_source()
+                try:
+                    source_methods = self.select_source_method(system, arg)
+                except NoSupportedSourcesFound as exception:
+                    # if no sources are found in the system for this
+                    # particular query, jump to the next one without
+                    # stopping execution
+                    LOG.error(exception, exc_info=debug)
+                    continue
+                for source_method, speed_score in source_methods:
+                    if source_methods_store.has_been_called(source_method):
+                        # we want to avoid repeating calls to the same
+                        # source method if several arguments with that
+                        # method are provided
+                        if source_methods_store.get_status(source_method):
+                            # if the previous call was successful, we do
+                            # not need to query the same method again
+                            break
+                        else:
+                            # if the previous call threw an error, let's
+                            # try a different source
+                            continue
+                    source_info = source_information_from_method(
+                            source_method)
+                    start_time = time.time()
+                    LOG.info(f"Performing query on system {system.name}")
+                    LOG.debug("Running %s and speed index %d",
+                              source_info, speed_score)
+                    # we update last_level if the the arg has a func attribute
+                    # and valid sources to query
+                    last_level = arg.level
```

Comment:
It should be moved before the sources are selected (line 212 for example) otherwise the issue still appears. I got it with the command
`cibyl --jobs pep8 --ip-version 4` and zuul and jenkins systems.